### PR TITLE
docs: add rough release process

### PR DESCRIPTION
**Pull request by chombourger**
Signed-off-by: Cedric Hombourger <cedric.hombourger@siemens.com>

### CODE REVIEW

File: docs/dev.rst [hunk 1]
```diff
@@ -45,21 +45,53 @@ and required Python packages and MTDA may be installed with::
...
++------+-------------------------------------------+------------+
+| -1w  | Collect ``Tested-by:`` tags               |            | 
++------+-------------------------------------------+------------+
+| -1w  | Draft ``debian/changelog``                |            |
++------+-------------------------------------------+------------+
+| -1w  | Use a fixed ``SRCREV`` in ``mtda_git.bb`` |            |
++------+-------------------------------------------+------------+
+| -1w  | Create -rc1 tag                           | v0.16-rc1  |
++------+-------------------------------------------+------------+
+|  0d  | Create release tag                        | v0.16      |
```
Comment (vj-kumar) [inline docs/dev.rst:90]: Wondering if we should add anything regarding artifacts. We are shipping some to sites. I guess automatically handled by CI during release?

File: docs/dev.rst [hunk 2]
```diff
@@ -45,21 +45,53 @@ and required Python packages and MTDA may be installed with::
...
++------+-------------------------------------------+------------+
+|  0d  | Move open issues to next milestone        |            |
++------+-------------------------------------------+------------+
+|  0d  | Send release announcement                 |            |
++------+-------------------------------------------+------------+
+|  0d  | Create PR for version bump                |            |
```
Comment (vj-kumar) [inline docs/dev.rst:96]: Basically we need python version bump + reverting to autorev

Comment (vj-kumar) [reply docs/dev.rst:96]: + dev changelog

File: docs/dev.rst (hunk 1 above)
Comment (chombourger) [reply docs/dev.rst]: Added a small paragraph below the table

**Pull request merged**